compute_tools/src/rsyslog.rs

```diff
@@ -75,3 +79,69 @@ pub fn configure_audit_rsyslog(
 
     Ok(())
 }
+
+#[instrument(skip_all)]
+fn pgaudit_gc_main_loop(log_directory: String) {
+    info!("running pgaudit GC main loop");
+    loop {
+        // Check log_directory for old pgaudit logs and delete them.
+        // New log files are checked every 5 minutes, as set in pgaudit.log_rotation_age
+        // Find files that were not modified in the last 15 minutes and delete them.
+        // This should be enough time for rsyslog to process the logs and for us to catch the alerts.
+        //
+        // In case of a very high load, we might need to adjust this value and pgaudit.log_rotation_age.
+        //
+        // TODO: add some smarted logic to delete the files that are fully streamed according to rsyslog
+        // imfile-state files, but for now just do a simple GC to avoid filling up the disk.
+        let _ = Command::new("find")
+            .arg(&log_directory)
+            .arg("-name")
+            .arg("audit*.log")
+            .arg("-mmin")
+            .arg("+15")
+            .arg("-delete")
+            .output()
+            .expect("Failed to execute find");
+
+        // also collect the metric for the size of the log directory
+        fn get_log_files_size(path: &Path) -> io::Result<u64> {
+            let mut total_size = 0;
+
+            for entry in fs::read_dir(path)? {
+                let entry = entry?; // unwrap entry
+                let entry_path = entry.path();
+
+                // Check if the entry is a file and ends with "log"
+                if entry_path.is_file() && entry_path.to_string_lossy().ends_with("log") {
+                    total_size += entry.metadata()?.len();
+                }
+            }
+
+            Ok(total_size)
+        }
+
+        let log_directory_size =
+            get_log_files_size(Path::new(&log_directory)).unwrap_or_else(|e| {
+                error!("Failed to get log directory size: {}", e);
+                0
+            });
+        crate::metrics::AUDIT_LOG_DIR_SIZE.set(log_directory_size as f64);
+
+        std::thread::sleep(Duration::from_secs(60));
+    }
+}
+
+// launch pgaudit GC thread to clean up the old pgaudit logs stored in the log_directory
+// pgaudit can rotate files, but it doesn't delete the old files
+// so we need to clean up the old files periodically
+pub fn launch_pgaudit_gc(log_directory: String) -> thread::JoinHandle<()> {
+    let runtime = tokio::runtime::Handle::current();
+    thread::Builder::new()
+        .name("compute-pg-audit-gc".into())
+        .spawn(move || {
+            let _rt_guard = runtime.enter();
```

Comment:
I'd suggest making this async (using tokio::time::sleep and so) as there's a general shift towards all-things-async in compute_ctl